src/rgw/rgw_pubsub.h

```diff
@@ -335,19 +335,15 @@ WRITE_CLASS_ENCODER(rgw_pubsub_s3_event)
 // setting a unique ID for an event based on object hash and timestamp
 void set_event_id(std::string& id, const std::string& hash, const utime_t& ts);
 
-struct rgw_pubsub_sub_dest {
-  std::string bucket_name;
-  std::string oid_prefix;
+struct rgw_pubsub_dest {
   std::string push_endpoint;
   std::string push_endpoint_args;
   std::string arn_topic;
   bool stored_secret = false;
   bool persistent = false;
 
   void encode(bufferlist& bl) const {
-    ENCODE_START(5, 1, bl);
-    encode(bucket_name, bl);
-    encode(oid_prefix, bl);
+    ENCODE_START(6, 1, bl);
```

Comment:
won't v5 decoders fail due to the removed fields? if so, then compat would need to go from 1 to 6

but this will interfere with rolling upgrades - if an upgraded radosgw/admin writes an object with v6, then v5 decoders won't be able to use them

maybe consider encoding two empty strings instead of bucket_name/oid_prefix to preserve compatibility? an empty string is just encoded as a 0 for the size